packages/api/src/middleware/auth.js

```diff
@@ -11,74 +19,70 @@ function authFactory(params) {
   return async (req, res, next) => {
     // must have either an API key (starts with 'Bearer') or a JWT token
     const authToken = req.headers.authorization;
+    const { tokenType, tokenValue } = parseAuthToken(authToken);
+    const basicUser = basicAuth.parse(authToken);
     let user;
     let tokenObject;
     let userId;
 
-    if (!authToken) {
-      throw new ForbiddenError(`no token object ${authToken} found`);
-    } else if (authToken.startsWith("Bearer")) {
-      tokenObject = await req.store.get(`api-token/${req.token}`);
-      if (!tokenObject) {
-        throw new ForbiddenError(`no token object ${authToken} found`);
+    if (!tokenType) {
+      throw new ForbiddenError(`no authorization header provided`);
+    } else if (["Bearer", "Basic"].includes(tokenType)) {
+      const isBasic = tokenType === "Basic";
+      const tokenId = isBasic ? basicUser?.pass : req.token;
+      if (!tokenId) {
+        throw new ForbiddenError(`no authorization token provided`);
+      }
+      tokenObject = await db.apiToken.get(tokenId);
+      const matchesBasicUser = tokenObject?.userId === basicUser?.name;
+      if (!tokenObject || (isBasic && !matchesBasicUser)) {
+        throw new ForbiddenError(`no token object ${tokenValue} found`);
```

Comment:
Nice, this will finally get rid of the really awkward `no token object undefined found`